main.tf

```diff
@@ -263,11 +263,10 @@ resource "aws_s3_bucket_replication_configuration" "default" {
       }
 
       dynamic "source_selection_criteria" {
-        for_each = try(rule.value.source_selection_criteria.sse_kms_encrypted_objects.enabled, null) == null ? [] : [rule.value.source_selection_criteria.sse_kms_encrypted_objects.enabled]
-
+        for_each = try(rule.value.source_selection_criteria.sse_kms_encrypted_objects.status, "") == "Enabled" ? [1] : []
```

Comment:
We want to track the Terraform resource as closely as possible, which includes being able to set the status to "Disabled". The current solution allows this, but unfortunately uses the object key "enabled" instead of "status". 

Suggestion:

```hcl
for_each = try(rule.value.source_selection_criteria.sse_kms_encrypted_objects.status, null) == null ? [] : [rule.value.source_selection_criteria.sse_kms_encrypted_objects.status]
```

Which allows a rule like:
```hcl
{
  source_selection_criteria = {
    sse_kms_encrypted_objects  = {
      status = "Disabled"
    }
  }
}
```

Except that would be a breaking change, so you need to support both `status` and `enabled`. Or you can just leave this wart as is, and configure using

```hcl
{
  source_selection_criteria = {
    sse_kms_encrypted_objects  = {
      enabled = "Enabled"
    }
  }
}
```

Reply:
Yup the point of this PR was because it wasn't clear that this module wasn't in-sync with the underlying Terraform resources and it wasn't intuitive to quickly go in a know to set `enabled = "Enabled"` since that wasn't mentioned anywhere. Especially that `enabled` is a bool anywhere else in the repo. But its okay we can leave it like this :man_shrugging: